File: leetcode/easy/UniqueMorseCodeWords.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        mtable = [
            '.-',
            '-...',
            '-.-.',
            '-..',
            '.',
            '..-.',
            '--.',
            '....',
            '..',
            '.---',
            '-.-',
            '.-..',
            '--',
            '-.',
            '---',
            '.--.',
            '--.-',
            '.-.',
            '...',
            '-',
            '..-',
            '...-',
            '.--',
            '-..-',
            '-.--',
            '--..'
        ]
        codes = set()
        for word in words:
            code = ''
            for c in word:
                # 97: ord('a')
                code += mtable[ord(c) - 97]
            codes.add(code)
        return len(codes)

File: leetcode/easy/UniqueMorseCodeWords.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] mTable = new String[]{
            ".-", "-...", "-.-.", "-..", ".",
            "..-.", "--.", "....", "..", ".---",
            "-.-", ".-..", "--", "-.", "---",
            ".--.", "--.-", ".-.", "...", "-",
            "..-", "...-", ".--", "-..-", "-.--",
            "--.."
        };
        Set<String> codes = new HashSet<>();
        for (String word : words) {
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < word.length(); i++) {
                int idx = (int)(word.charAt(i) - 'a');
                sb.append(mTable[idx]);
            }
            codes.add(sb.toString());
        }
        return codes.size();
    }
}
